### ISSUE TITLE
Substrait extension fails to load on 0.5.1

### ISSUE THREAD

**Issue by lupko** [duckdb/duckdb#4873]
### What happens?

While installing and loading `substrait`, the load extension fails with:

`duckdb.IOException: IO Error: Extension "/home/user/.duckdb/extensions/v0.5.1/linux_amd64_gcc4/substrait.duckdb_extension" could not be loaded: /home/user/.duckdb/extensions/v0.5.1/linux_amd64_gcc4/substrait.duckdb_extension: undefined symbol: _ZTIN6duckdb9ExceptionE`

The error does not occur on 0.5.0.

Note: wanted to try this on latest prerelease but can't seem to drag it from PyPi (`pip install --pre duckdb==0.5.2.dev439` fails & the `pip duckdb upgrade --pre` does nothing).

### To Reproduce

pip install duckdb=0.5.1
python -c 'import duckdb; con = duckdb.connect(); con.install_extension("substrait"); con.load_extension("substrait")'

### OS:

Linux

### DuckDB Version:

0.5.1

### DuckDB Client:

Python

### Full Name:

Lubo Slivka

### Affiliation:

GoodData Inc.

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mause** [comment]
Can you try `pip install duckdb --upgrade --pre` instead?

**Comment by lupko** [comment]
no game with that, still installs 0.5.1

```
~> pip install duckdb --upgrade --pre
Defaulting to user installation because normal site-packages is not writeable
Collecting duckdb
  Using cached duckdb-0.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.4 MB)
Requirement already satisfied: numpy>=1.14 in /usr/lib64/python3.10/site-packages (from duckdb) (1.22.0)
Installing collected packages: duckdb
Successfully installed duckdb-0.5.1
~> pip install duckdb --upgrade --pre
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: duckdb in ./.local/lib/python3.10/site-packages (0.5.1)
Requirement already satisfied: numpy>=1.14 in /usr/lib64/python3.10/site-packages (from duckdb) (1.22.0)
```

**Comment by Mytherin** [comment]
Is this problem limited to the substrait extension, or do other extensions have the same problem?

**Comment by lupko** [comment]
tried with `extensions = ['fts', 'httpfs', 'icu', 'json', 'parquet', 'postgres_scanner', 'sqlite_scanner', 'tpcds', 'tpch', 'substrait']`

only substrait fails during load

**Comment by samansmink** [comment]
see https://github.com/duckdb/duckdb/issues/4595, should be resolved for v0.6.0
